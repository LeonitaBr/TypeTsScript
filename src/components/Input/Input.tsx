import React, {memo} from 'react';
import {TextInput} from 'react-native';
import styles from './style';

const InputComponent = () => {
  const [text, onChangeValue] = React.useState('Useless Text');
  return (
    <TextInput
      style={styles.input}
      onChangeText={value => onChangeValue(value)}
      value={text}
      placeholder="useless placeholder"
      keyboardType="numeric"
    />
  );
};
export default memo(InputComponent);
