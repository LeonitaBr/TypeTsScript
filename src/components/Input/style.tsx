import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  input: {
    height: 40,
    borderWidth: 1,
    padding: 10,
  },
});
export default styles;
