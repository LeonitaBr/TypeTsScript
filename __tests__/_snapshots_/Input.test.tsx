import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../../src/components/Input/Input';

describe('Some component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
