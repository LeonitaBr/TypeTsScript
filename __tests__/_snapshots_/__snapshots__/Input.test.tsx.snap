// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some component renders correctly 1`] = `
<TextInput
  keyboardType="numeric"
  onChangeText={[Function]}
  style={
    Object {
      "borderWidth": 1,
      "height": 40,
      "padding": 10,
    }
  }
/>
`;
